sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -94,7 +103,11 @@ public void AddLink(string traceparent, string tracestate, IDictionary<string, s
 
         public void Start()
         {
-            _activityAdapter?.Start();
+            bool? started = _activityAdapter?.Start();
+            if (started.GetValueOrDefault() && _suppressNestedClientScopes)
```

Comment:
nit - for the default case, it might be a bit more efficient to check `_suppressNestedClientScopes` first